manifests/share.pp

```diff
@@ -3,9 +3,9 @@
 # This class setups up a config file to be included into smb.conf
 #
 define winbind::share (
-  $ensure_setting  = present,
-  $settings_hash   = undef,
-  $smb_include_dir = $::winbind::smb_includes_dir,
+  Boolean       $ensure_setting  = lookup('winbind::share::ensure_setting'),
```

Comment:
Uh oh, no I'll fix that. According to the [doc](https://docs.puppet.com/puppet/5.0/lang_data_boolean.html), it only matches `true` or `false`, so I'm not sure how this wasn't failing in the unit testing.